src/coreclr/jit/morph.cpp

```diff
@@ -7119,13 +7119,29 @@ bool Compiler::fgCallHasMustCopyByrefParameter(GenTreeCall* callee)
                             // We only check this for calls with small numbers of arguments,
                             // as the analysis cost will be quadratic.
                             //
-                            if (varDsc->lvRefCnt(RCS_EARLY) == 1)
+                            const unsigned totalAppearances = varDsc->lvRefCnt(RCS_EARLY);
+                            const unsigned callAppearances  = (unsigned)varDsc->lvRefCntWtd(RCS_EARLY);
+                            assert(totalAppearances >= callAppearances);
+
+                            if (totalAppearances == 1)
                             {
                                 JITDUMP("... yes, arg is the only appearance of V%02u\n", lclNum);
                                 hasMustCopyByrefParameter = false;
                             }
+                            else if (totalAppearances > callAppearances)
+                            {
+                                // lvRefCntWtd tracks the number of appearances of the arg at call sites.
+                                // If this number doesn't match the regular ref count, there is
+                                // a non-call appearance, and we must be conservative.
+                                //
+                                JITDUMP("... no, arg has %u non-call appearance(s)\n",
+                                        totalAppearances - callAppearances);
+                            }
```

Comment:
Is it possible to see two arguments like:
```
GT_LCL_VAR a

GT_COMMA
  GT_ASG
    GT_LCL_VAR b
    GT_ADDR
      GT_LCL_VAR a
  GT_LCL_VAR b
```

and what do the ref counts look like in this case?

Reply:
The counts should be 1 call, 1 non-call for both `a` and `b`. The counting logic is in `UpdateEarlyRefCountForImplicitByRef`.